lib/galaxy/datatypes/tabular.py

```diff
@@ -788,45 +788,41 @@ def sniff( self, filename ):
             - We will only check that up to the first 5 alignments are correctly formatted.
         """
         try:
-            fh = compression_utils.get_fileobj(filename, gzip_only=True)
-            count = 0
-            while True:
-                line = fh.readline()
-                line = line.strip()
-                if not line:
-                    break  # EOF
-                if line:
-                    line_pieces = line.split('\t')
-                    if len(line_pieces) != 22:
-                        return False
-                    try:
-                        if long(line_pieces[1]) < 0:
-                            raise Exception('Out of range')
-                        if long(line_pieces[2]) < 0:
-                            raise Exception('Out of range')
-                        if long(line_pieces[3]) < 0:
-                            raise Exception('Out of range')
-                        int(line_pieces[4])
-                        int(line_pieces[5])
-                        # can get a lot more specific
-                    except ValueError:
-                        fh.close()
-                        return False
-                    count += 1
-                    if count == 5:
-                        break
-            if count > 0:
-                fh.close()
-                return True
-        except:
+            with compression_utils.get_fileobj(filename, gzip_only=True) as fh:
+                count = 0
+                while True:
+                    line = fh.readline()
+                    line = line.strip()
+                    if not line:
+                        break  # EOF
+                    if line:
+                        line_pieces = line.split('\t')
+                        if len(line_pieces) != 22:
+                            return False
+                        try:
+                            if long(line_pieces[1]) < 0:
+                                raise Exception('Out of range')
+                            if long(line_pieces[2]) < 0:
+                                raise Exception('Out of range')
+                            if long(line_pieces[3]) < 0:
+                                raise Exception('Out of range')
+                            int(line_pieces[4])
+                            int(line_pieces[5])
+                            # can get a lot more specific
+                        except ValueError:
+                            return False
+                        count += 1
+                        if count == 5:
+                            break
+                if count > 0:
+                    return True
+        except Exception:
             pass
```

Comment:
You can remove the external `try/except`, exceptions in a sniffer are already caught in `galaxy.datatypes.sniff.guess_ext()` and are equivalent to the following `return False`.